stripe/src/main/java/com/stripe/android/CustomerSession.java

```diff
@@ -133,6 +133,21 @@ public static void endCustomerSession() {
         clearInstance();
     }
 
+    /**
+     * End any async calls in process and will not invoke callback listeners.
+     * It will not clear the singleton instance of a {@link CustomerSession} so it can be
+     * safely used when a view is being removed/destroyed to avoid null pointer exceptions
+     * due to async operation delay.
+     * No need to call {@link CustomerSession#initCustomerSession(EphemeralKeyProvider)} again
+     * after this operation.
+     */
+    public static void cancelCallbacks(){
```

Comment:
- add space before `{`
- can you mark the `mThreadPoolExecutor` instance variable as final (line 82)?
- move this to below the `clearInstance()` method
- call this method from `clearInstance()`

```
static void clearInstance() {
  cancelCallbacks();
  mInstance = null;
}
```

Reply:
Hi, I will get back to you with requested changes.

Reply:
Thanks!